components/org.wso2.carbon.identity.api.server.organization.management/org.wso2.carbon.identity.api.server.organization.management.v1/src/main/java/org/wso2/carbon/identity/api/server/organization/management/v1/service/OrganizationManagementService.java

```diff
@@ -653,6 +830,113 @@ private SharedApplicationsResponse createSharedApplicationsResponse(List<SharedA
+                    return attribute;
+                }).collect(Collectors.toList());
+        discoveryAttributes.setAttributes(attributes);
+        return discoveryAttributes;
+    }
+
+    private List<OrgDiscoveryAttribute> getOrgDiscoveryAttributesFromPostRequest(OrganizationDiscoveryPostRequest
+                                                                                         discoveryPostRequest) {
+
+        return Optional.ofNullable(discoveryPostRequest.getAttributes()).orElse(Collections.emptyList())
+                .stream()
+                .map(attribute -> {
+                    OrgDiscoveryAttribute orgDiscoveryAttribute = new OrgDiscoveryAttribute();
+                    orgDiscoveryAttribute.setType(attribute.getType());
+                    orgDiscoveryAttribute.setValues(attribute.getValues());
+                    return orgDiscoveryAttribute;
+                }).collect(Collectors.toList());
+    }
+
+    private List<OrgDiscoveryAttribute> getOrgDiscoveryAttributesFromPutRequest(OrganizationDiscoveryAttributes
+                                                                                        discoveryAttributes) {
+
+        return Optional.ofNullable(discoveryAttributes.getAttributes()).orElse(Collections.emptyList())
+                .stream()
+                .map(attribute -> {
+                    OrgDiscoveryAttribute orgDiscoveryAttribute = new OrgDiscoveryAttribute();
+                    orgDiscoveryAttribute.setType(attribute.getType());
+                    orgDiscoveryAttribute.setValues(attribute.getValues());
+                    return orgDiscoveryAttribute;
+                }).collect(Collectors.toList());
+    }
+
+    private void handleNotImplementedCapabilities(String filter, Integer offset, Integer limit) {
+
+        if (limit != null || offset != null) {
+            Error error = OrganizationManagementEndpointUtil.getError(
+                    ERROR_CODE_PAGINATION_NOT_IMPLEMENTED.getCode(),
+                    ERROR_CODE_PAGINATION_NOT_IMPLEMENTED.getMessage(),
+                    ERROR_CODE_PAGINATION_NOT_IMPLEMENTED.getDescription());
+            throw new OrganizationManagementEndpointException(Response.Status.NOT_IMPLEMENTED, error);
+        }
+        if (filter != null) {
+            Error error = OrganizationManagementEndpointUtil.getError(
+                    ERROR_CODE_FILTERING_NOT_IMPLEMENTED.getCode(),
+                    ERROR_CODE_FILTERING_NOT_IMPLEMENTED.getMessage(),
+                    ERROR_CODE_FILTERING_NOT_IMPLEMENTED.getDescription());
+            throw new OrganizationManagementEndpointException(Response.Status.NOT_IMPLEMENTED, error);
+        }
+    }
+
+    private OrganizationMetadata getOrganizationMetadataResponse(Organization organization, List<OrgDiscoveryAttribute>
+            organizationDiscoveryAttributes) {
+
+        OrganizationMetadata organizationMetadata = new OrganizationMetadata();
```

Comment:
Can use builder pattern for this

Reply:
OrganizationMetadata is an autogenerated class. hence unable to modify the class to support the builder pattern